src/main/java/com/osiris/autoplug/client/managers/SyncFilesManager.java

```diff
@@ -26,68 +27,83 @@
 
 public class SyncFilesManager {
     private static List<File> lastFoldersToWatch = new ArrayList<>();
+
     public SyncFilesManager(SharedFilesConfig sharedFilesConfig) throws Exception {
+        clearPreviousWatchers();
+        List<File> foldersToWatch = initializeFoldersToWatch(sharedFilesConfig);
+        List<File> filesToSendTo = initializeFilesToSendTo(sharedFilesConfig);
+        Consumer<FileEvent> onFileChangeEvent = defineFileChangeEventHandler(filesToSendTo);
+        setUpDirectoryWatchers(foldersToWatch, onFileChangeEvent);
+    }
+
+    private void clearPreviousWatchers() throws Exception {
         for (File folder : lastFoldersToWatch) {
             DirWatcher dirWatcher = DirWatcher.get(folder, false);
             dirWatcher.removeAllListeners(true);
             dirWatcher.close();
         }
+    }
 
+    private List<File> initializeFoldersToWatch(SharedFilesConfig sharedFilesConfig) throws Exception {
         List<File> foldersToWatch = new ArrayList<>();
-        for (String pathAsString :
-                sharedFilesConfig.copy_from.asStringList()) {
+        for (String pathAsString : sharedFilesConfig.copy_from.asStringList()) {
             if (pathAsString.startsWith("./"))
                 foldersToWatch.add(FileManager.convertRelativeToAbsolutePath(pathAsString));
             else
                 throw new Exception("Wrongly formatted or absolute path: " + pathAsString);
         }
+        return foldersToWatch;
+    }
 
+    private List<File> initializeFilesToSendTo(SharedFilesConfig sharedFilesConfig) throws Exception {
         List<File> filesToSendTo = new ArrayList<>();
-        //List<String> ipsToSendTo = new ArrayList<>();
```

Comment:
keep these comments since they are relevant for future feature implementations

Reply:
Thank you so much for reviewing my pull request!
I will look into the code again considering the changes you suggested!